docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/builder/DockerBuilderPublisher.java

```diff
@@ -230,19 +225,20 @@ public String invoke(File f, VirtualChannel channel) throws IOException, Interru
                             listener.getLogger().println("Docker Build : building tag " + tag);
 
                             try {
-                                Iterable<EventStreamItem> response = getClient().buildImageCmd(f)
-                                        .withTag(tag)
-                                        .exec().getItems();
-
-                                for (EventStreamItem item : response) {
-                                    String text = item.getStream();
-                                    if (text != null) {
-                                        listener.getLogger().append(text);
-                                        if (text.startsWith("Successfully built ")) {
-                                            imageId = StringUtils.substringBetween(text, "Successfully built ", "\n").trim();
+                                BuildImageResultCallback resultCallback = new BuildImageResultCallback() {
+                                    public void onNext(BuildResponseItem item) {
+                                        String text = item.getStream();
+                                        if (text != null) {
+                                            listener.getLogger().append(text);
```

Comment:
Could you try this https://github.com/jenkinsci/docker-plugin/pull/300/files#r37809619 trick for builder/publisher?